packages/repeater/src/request-runner/protocols/HttpRequestRunner.ts

```diff
@@ -0,0 +1,244 @@
+import { Request } from '../Request';
+import { RequestRunner } from '../RequestRunner';
+import { RequestRunnerOptions } from '../RequestRunnerOptions';
+import { Response } from '../Response';
+import { Protocol } from '../../models';
+import { Logger } from '@secbox/core';
+import request from 'request-promise';
+import { Response as IncomingResponse } from 'request';
+import { SocksProxyAgent } from 'socks-proxy-agent';
+import { inject, injectable } from 'tsyringe';
+import { parse as contentTypeParse } from 'content-type';
+import { parse } from 'url';
+import http, { OutgoingMessage } from 'http';
+import https, { AgentOptions } from 'https';
+
+@injectable()
+export class HttpRequestRunner implements RequestRunner {
+  private readonly proxy?: SocksProxyAgent;
+  private readonly httpAgent?: http.Agent;
+  private readonly httpsAgent?: https.Agent;
+  private readonly maxContentLength: number;
+
+  get protocol(): Protocol {
+    return Protocol.HTTP;
+  }
+
+  constructor(
+    @inject(RequestRunnerOptions)
+    private readonly options: RequestRunnerOptions,
+    @inject(Logger)
+    private readonly logger?: Logger
```

Comment:
IMO there is no need to mark it as optional. Wdyt?

```suggestion
    private readonly logger: Logger
```